pkg/userdata/coreos/provider.go

```diff
@@ -198,9 +214,13 @@ systemd:
         TimeoutStartSec=5min
         CPUAccounting=true
         MemoryAccounting=true
+{{- if .NodeHttpProxy }}
+        Environment=KUBELET_IMAGE=docker://{{ .NodeImageRegistry }}/machine-controller/hyperkube-amd64:{{ .HyperkubeImageTag }}
```

Comment:
Instead of re-using this magic variable which other than that adds an insecure-registry, can we add an explicit config option for the registry+namespace+name of the hyperkube image in the CoreOS option?

Reply:
I'll add a flag to pass through the image for hyperkube & the pause container

Reply:
Done with https://github.com/kubermatic/machine-controller/pull/559/commits/649786bd2d98d7c54b81975417e047a9a654ab42